app/models/share.rb

```diff
@@ -9,6 +9,8 @@ class Share < ApplicationRecord
   belongs_to :user
   belongs_to :work
 
+  after_create -> { Notifier.publish(Notifier::SHARE_ADDED, share: self, work:) }
```

Comment:
Do you need to provide the work?

Reply:
no you don't, you can get it from the share... but if you don't pass in the work, these lines will cause trouble, because it tries to get the work from the params directly and then uses that to get the user: 

https://github.com/sul-dlss/hungry-hungry-hippo/blob/main/app/mailers/works_mailer.rb#L5
https://github.com/sul-dlss/hungry-hungry-hippo/blob/main/app/mailers/application_mailer.rb#L8-L9

We could skip the `before_action` for just this mailer method and set the work and user separately via the share, but that seemed like it would be harder to understand by a developer later than just passing in the work too.